Detect Jenkins CI

fdlk opened the pull request:
The GitHub repo slug can be determined from `GIT_URL`.
If there are multiple repo's, you can expect GitHub to be in `GIT_URL_1`.

For PRs, the pull request ID is in `CHANGE_ID`.
For other builds, the latest commit is in `GIT_COMMIT`.

Fixes #488

AshesITR commented:
Hi @fdlk. are you still interested in this?
If you add a NEWS bullet, we can merge this I think.

AshesITR commented:
Hi @fdlk. are you still interested in this?
If you add a NEWS bullet, we can merge this I think.

fdlk commented:
@AshesITR yay! nice! I've rebased onto the current master and added a NEWS bullet. Build still works except in macOS. A cursory look makes me think that's not related to the PR, would you agree?

AshesITR commented:
Thanks a lot. I've restarted the macOS build, it indeed seems spurious.
